skills/inferskills.py

```diff
@@ -0,0 +1,111 @@
+import numpy as np
+import json
+
+
+class InferSkills:
+  def __init__(self):
+    self.vals = {}
+    with open('skillsets_labels.tsv', 'r') as f:
+      for line in f:
+        items = line.split('\t')
+        k = items[1]
+        v = set(items[2].replace('\n', '').lower().split(', '))
+        self.vals[k] = v
+  
+  def render_scores(self, data, scores, keywords):
+    for i in data:
+      for k in self.vals.keys():
+        for v in self.vals[k]:
+          if len(v.split(" ")) == 1:
+            if v in i.split(" "):
+              scores[k] += 1
+              if v in keywords.keys():
+                keywords[v] += 1
+              else:
+                keywords[v] = 1
+          else:
+            if v in i:
+              scores[k] += 1
+              if v in keywords.keys():
+                keywords[v] += 1
+              else:
+                keywords[v] = 1
+  
+  def infer_exp(self, p, e, return_scores = False):
+    if "role" not in e.keys() or e["role"] == None:
```

Comment:
keys() yields a list, which loses you the computational advantage of a dictionary look-up. Instead, the better way to do this is either:
(a) if 'role' in e and e['role']:
(b) if e.get('role',None):